emanote/src/Emanote/Model/Note.hs

```diff
@@ -316,7 +320,19 @@ parseNote scriptingEngine pluginBaseDir r src@(_, fp) s = do
         parseNoteMarkdown scriptingEngine pluginBaseDir fp s
       R.LMLRoute_Org _ -> do
         parseNoteOrg s
-  pure $ mkNoteWith r (Just src) doc meta errs
+  let metaWithDateFromPath = case P.parse dateParser mempty (takeFileName fp) of
+        Left _ -> meta
+        Right date -> SData.modifyAeson (pure "date") (const $ Just $ Aeson.String date) meta
```

Comment:
If the note already has `date` metadata, I'd think we should use that over the inferred one.

Reply:
Good point, I updated the PR to preserve existing metadata.